### PR TITLE
Add DVS signature verification

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**

### CODE REVIEW

File: pkg/dvs/dvs.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	}
+
+	verifyEndpoint, err := url.Parse("/dvs/verify")
+	if err != nil {
+		return InvalidSignature, err
+	}
+
+	verifyURL := baseAddr.ResolveReference(verifyEndpoint)
+
+	req, err := http.NewRequest("POST", verifyURL.String(), bytes.NewReader(reqBodyJSON))
```
Comment (VMitov) [inline pkg/dvs/dvs.go]: Expose the http client trough the Client.

File: pkg/dvs/dvs.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+// New returns new DVS client.
+func New(ctx context.Context, mfa *mfa.Client) *Client {
+	keys := oidc.NewRemoteKeySet(ctx, fmt.Sprintf("%v/dvs/jwks", mfa.Issuer()))
+	return &Client{mfa: mfa, Keys: keys}
+}
+
+// VerifySignature sends signature for verification to the DVS (designated verifier scheme) service and verifies the received response.
+func (c *Client) VerifySignature(ctx context.Context, dvsReq *VerifyRequest) (VerificationResult, error) {
```
Comment (VMitov) [inline pkg/dvs/dvs.go]: Separate the DVS business logic from the HTTP request/response cycle.

File: pkg/dvs/dvs.go [hunk 3]
```diff
@@ -64,79 +64,130 @@ func (r VerificationResult) String() string { return verificationResultText[r] }
 
 // Client is a client used for DVS verification.
 type Client struct {
-	mfa  *mfa.Client
-	Keys oidc.KeySet
+	mfa        *mfa.Client
+	Keys       oidc.KeySet
+	HTTPClient *http.Client
```
Comment (VMitov) [inline pkg/dvs/dvs.go]: Should this be exported?

Comment (MiroslavGatsanoga) [reply pkg/dvs/dvs.go]: Nope.

**Pull request merged**